### ISSUE TITLE
Possible false-positive for `prefer_expression_function_bodies`

### ISSUE THREAD

**Issue by incendial** [dart-lang/linter#4221]
**Describe the issue**
Hi, consider the following example:

```dart
@override
class SomeVisitor {
  @override
  void visitBlockFunctionBody(BlockFunctionBody node) {
    return;
  }
}
```

The rule suggests to change the return statement into =>
<img width="628" alt="Screenshot 2023-03-26 at 19 42 26" src="https://user-images.githubusercontent.com/21116742/227786970-9146da4c-74fd-4c08-a9c0-9a9e7ac1aedb.png">

but in this case I can only replace it with `=> null` and then another rule starts to complain
<img width="492" alt="Screenshot 2023-03-26 at 19 43 49" src="https://user-images.githubusercontent.com/21116742/227787067-ba0988c4-3825-4dd6-850d-1d4d925afbd5.png">

Is it a false positive or what is the expected proper fix for this case?

**Comment by srawlins** [comment]
+1; I'd say false positive.

**Comment by pq** [comment]
💯 

Thanks for the report!

**Comment by incendial** [comment]
You're welcome! ☺️
